### PR TITLE
Doxygen->Breathe->Sphinx MLL docs 

### PR DESCRIPTION
**Pull request by brtnfld**
Replace MLL docs with Doxygen-generated documentation.

- [x] **Look into this later:** Need to resolve the Doxygen issue with MACRO and undefined identifiers, for example: WARNING: cpp:identifier reference target not found: cgsize_t
- [x] Fix errors like : ERROR: Unknown interpreted text role "version".
- [x] Update download.rst
- [x]  Update/upload minutes
- [x] Fix common errors: time.rst:202: WARNING: Pygments lexer name 'sids' is not known
- [x] Remove all links to Chris' NASA email
- [x] Remove all links to Bob Bush' PW email
- [x] Fix the TODO:ADDLINK cases when the modern site becomes the default, need to work around: https://github.com/breathe-doc/breathe/discussions/982#discussioncomment-10507250

### CODE REVIEW

**Comment by brtnfld** [comment]
@MicK7, could you review the current webpage built with this fork? I switched to using  sphinx_book_theme. I like the right-hand side menu, especially for navigating the APIs. I disabled the dark mode because I don't think it looks all that great, and I'm not a fan of the purple highlight with orange text. Other than that, I think it looks good.

I tried most of the other themes, and they either had an issue rendering the doxygen/breathe output or would not let me customize it without needing a lot of CSS overrides, and I'm not that versed in CSS.

I added the CGNS repo as a submodule, so you should use the following:

`git clone --recurse-submodules https://github.com/brtnfld/cgns-modern.github.io.git`

Then in `CGNS-ReST-site`, you can do `make html`

I still need to move over some particle content from the HTML pages.

**Comment by MicK7** [comment]
> other themes, and they either had an issue rendering the doxygen/breathe output or would not let me customize it without needing a lot of CSS overrides, and



> @MicK7, could you review the current webpage built with this fork? I switched to using sphinx_book_theme. I like the right-hand side menu, especially for navigating the APIs. I disabled the dark mode because I don't think it looks all that great, and I'm not a fan of the purple highlight with orange text. Other than that, I think it looks good.
> 
> I tried most of the other themes, and they either had an issue rendering the doxygen/breathe output or would not let me customize it without needing a lot of CSS overrides, and I'm not that versed in CSS.
> 
> I added the CGNS repo as a submodule, so you should use the following:
> 
> `git clone --recurse-submodules https://github.com/brtnfld/cgns-modern.github.io.git`
> 
> Then in `CGNS-ReST-site`, you can do `make html`
> 
> I still need to move over some particle content from the HTML pages.

Thanks. I will look into it as soon as possible.
Like you I would prefer a light color theme as the original website and not a dark one.

**Comment by brtnfld** [comment]
Thanks. I'm also not sure the site map is needed on the landing page, but I'm not sure how to remove it without also removing the site map in the right-hand side panel.

**Comment by MicK7** [comment]
> Thanks. I'm also not sure the site map is needed on the landing page, but I'm not sure how to remove it without also removing the site map in the right-hand side panel.

I think it can be hidden by declaring the toctree hidden 
```rest
.. toctree::
  :maxdepth: 4
  :hidden:

  general
  ...
```

The new theme is pretty nice.

**Comment by brtnfld** [comment]
Yes, that worked to hide the TOC. I'm also not sure if the Main Topics Table is needed, as it partially repeats the RHS menu and is somewhat disconnected from the topics in the RHS menu. I'm also still determining about the news section as it tends to get behind and needs to be updated.

**Comment by MicK7** [comment]
Maybe an item should be added to the TODO list:
- [x] Update Minutes

**Comment by brtnfld** [comment]
@MicK7, I'm close to calling this "good enough, for now," with just a few tasks remaining. We should switch to this website as the main webpage, archive the old site (and ensure the search engines do not prioritize it over the current one), and archive the old webpage repository. This task would be nice to have before the release.

**Comment by MicK7** [comment]
Sounds good to me. Do we create a cgns-archive.github.io for the old site ?

**Comment by brtnfld** [comment]
> Sounds good to me. Do we create a cgns-archive.github.io for the old site ?

We can rename the cgns.github.io to cgns-archive.github.io, then rename cgns-modern.github.io to cgns.github.io.

File: CGNS-ReST-site/source/standard/MLL/api/general_remarks.rst [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+****************************************
+    Typedefs
+****************************************
+
+Beginning with CGNS-3.1.0, two new typedef variables have been introduced to support 64-bit mode. The *cglong_t* typedef
+is always a 64-bit integer, and ``cgsize_t`` will be either a 32-bit or 64-bit integer depending on how the library was built.
+Many of the C functions in the MLL have been changed to use ``cgsize_t`` instead of *int* in the arguments. These functions
+include any that may exceed the 2Gb limit of an int, e.g. zone dimensions, element data, boundary conditions, and connectivity.
+In Fortran, all integer data is taken to be *INTEGER\*4* for 32-bit and *INTEGER\*8* for 64-bit builds.
```
Comment (brtnfld) [inline CGNS-ReST-site/source/standard/MLL/api/general_remarks.rst]: @MicK7  I'm not entirely sure that the sentence is accurate or what specific data it refers to. INTEGERs are not promoted to 64-bit with enable 64-bit. I think it should be removed..

**Comment by brtnfld** [comment]
- [ ]  Add links back into the overview section

File: CGNS-ReST-site/source/standard/MLL/api/general_remarks.rst (hunk 1 above)
Comment (MicK7) [reply CGNS-ReST-site/source/standard/MLL/api/general_remarks.rst]: I agree let's remove it. The first time I read it I though it was for cgsize_t type but the "all integer" is completly wrong.

**Comment by brtnfld** [comment]
@MicK7 , I think I am done with my updates. The PR is getting large, so any more updates should be made to the new PR. 
I think it is in a state that it can replace the current one as the default.

**Comment by MicK7** [comment]
> @MicK7 , I think I am done with my updates. The PR is getting large, so any more updates should be made to the new PR. 
> I think it is in a state that it can replace the current one as the default.

I agree. Let's do the replacement and update through new PRs after to polish the result.

**Review by MicK7 — APPROVED** [review]

**Pull request merged**